### PR TITLE
Improve streaming ingestion completion timeout error message

### PR DESCRIPTION
**Pull request by AmatyaAvadhanula**
A streaming ingestion supervisor spec has the property: `completionTimeout` (defaults to 30 minutes), which is the total time to wait after the taskDuration has elapsed for the tasks to stop reading, create segments from intermediate persists, publish segments to deep storage, and finally hand them off from the peons to the historicals.

A task fails if the above operations are not completed before the completionTimeout has elapsed.

This PR adds steps for better debugging when such failures occur.

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java [hunk 1]
```diff
@@ -3271,7 +3271,9 @@ private void checkPendingCompletionTasks()
             log.warn("All tasks in group [%d] failed to publish, killing all tasks for these partitions", groupId);
           } else {
             log.makeAlert(
-                "No task in [%s] for taskGroup [%d] succeeded before the completion timeout elapsed [%s]!",
+                "No task in [%s] for taskGroup [%d] succeeded before the completion timeout elapsed [%s]! "
+                + "Please check the total time taken for each segment's creation and publish to deep storage, "
+                + "as well as the total time taken for segment handoff to the historicals.",
```
Comment (kfaraz) [inline indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java:3276]:
```suggestion
                + "Check metrics to see if the creation, publish or handoff of any segment is taking longer than usual.",
```

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java [hunk 2]
```diff
@@ -3271,7 +3271,9 @@ private void checkPendingCompletionTasks()
             log.warn("All tasks in group [%d] failed to publish, killing all tasks for these partitions", groupId);
           } else {
             log.makeAlert(
-                "No task in [%s] for taskGroup [%d] succeeded before the completion timeout elapsed [%s]!",
+                "No task in [%s] for taskGroup [%d] succeeded before the completion timeout elapsed [%s]! "
+                + "Check metrics and logs to see if the creation, publish or handoff of any segment is taking longer "
+                + "than usual.",
```
Comment (kfaraz) [inline indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java:3276]: Nit: break the line sooner
```suggestion
                + "Check metrics and logs to see if the creation, publish or handoff"
                + "  of any segment is taking longer than usual.",
```

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java [hunk 3]
```diff
@@ -3271,7 +3271,9 @@ private void checkPendingCompletionTasks()
             log.warn("All tasks in group [%d] failed to publish, killing all tasks for these partitions", groupId);
           } else {
             log.makeAlert(
-                "No task in [%s] for taskGroup [%d] succeeded before the completion timeout elapsed [%s]!",
+                "No task in [%s] for taskGroup [%d] succeeded before the completion timeout elapsed [%s]! "
+                + "Check metrics and logs to see if the creation, publish or handoff"
+                + "  of any segment is taking longer than usual.",
```
[the remark this answers is not in the document]
Comment (asdf2014) [reply indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java:3276]: nit: remove the extra blank char

**Review by asdf2014 — CHANGES_REQUESTED** [review]

**Review by asdf2014 — APPROVED** [review]

**Comment by AmatyaAvadhanula** [comment]
@kfaraz, @asdf2014 Thank you for the review!

**Pull request merged**